ext/handlers/filters/message/message.go

```diff
@@ -306,3 +306,15 @@ func TopicAction(msg *gotgbot.Message) bool {
 	return TopicEdited(msg) || TopicCreated(msg) ||
 		TopicClosed(msg) || TopicReopened(msg)
 }
+
+func MessageEntityPre(offset int, length int, language string) filters.Filter {
+	return func(ctx *gotgbot.CallbackContext) bool {
+		entities := ctx.Message.Entities
+		for _, entity := range entities {
+			if entity.Offset == offset && entity.Length == length && entity.Language == language {
```

Comment:
Do you have any example scenarios where one is able to predict both the offset and the length of an incoming message entity? I am not able to come up with one myself.

Additionally, this check is not asserting that the entity is a `pre` entity, as suggested by the method name.

Reply:
predicting both the offset and length of an incoming message entity might not always be feasible or practical, especially in real-time scenarios. However, if you're dealing with predefined message formats or templates, you could potentially predict these values based on the structure of your messages.

<u>**Here is one example to you understand proper**</u>
Let's say you have a system where users submit code snippets in a specific format, and you know that each code snippet starts at a fixed offset and has a fixed length. In this case, you can predict the offset and length of the code snippet entity based on the message format.

However, if the offset and length need to be dynamically determined based on user input or other factors, predicting them accurately might be challenging. In such cases, it's often better to rely on other attributes or patterns in the message content rather than trying to predict specific offsets and lengths.

Regarding the method name not asserting that the entity is a pre entity, you're correct. If you want to specifically filter preformatted code entities, you should include additional checks for the entity type.

Reply:
As you've noted yourself, this isnt feasible or practical. Which makes this function very unlikely to be used. Users that will hit the niche usecase you've mentioned can simply implement the filter themselves.

Given this, it sounds like we agree that this PR should not be merged.